### PR TITLE
docs(gateways): add doc on creating Brigade gateways

### LINKED ISSUE #227: How to work with cron jobs?
The docs or the announcement blog post mention that brigade is a good fit to run scheduled jobs. Is there an example of integration with Kubernetes cron jobs?

### PR DESCRIPTION
**Pull request by technosophos**
Closes #227

### CODE REVIEW

File: docs/topics/gateways.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  labels:
+    # 'heritage: brigade' is mandatory, and signals that this is a Brigade event.
+    heritage: brigade
+
+    # This should point to the Brigade project ID in which this event is to be
+    # executed
+    project: brigade-1234567890
+
+    # This MUST be a unique ID. Where possible, it SHOULD be a ULID
+    # Substituting a UUID is fine, though some sorting functions won't be as expected
```
Comment (mbarison) [inline docs/topics/gateways.md]: what is the purpose of sorting in this context?

File: docs/topics/gateways.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    build: 01C1R2SYTYAR2WQ2DKNTW8SH08
+
+    # Commit is the commitish/reference for any associated VCS repository. By
+    # default, this should be `master` for Git.
+    commit: master
+
+    # 'component: build' is REQUIRED and tells brigade to start a new build.
```
Comment (mbarison) [inline docs/topics/gateways.md]: What if I don't want to build stuff? Maybe I just want to schedule tests on an already deployed release.

File: docs/topics/examples/simple-event/simple-event.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    name: ${name}
+    labels:
+      heritage: brigade
+      project: ${project_id}
+      build: ${uuid}
+      commit: ${commit}
+      component: build
+  type: Opaque
+  data:
+    commit: $(echo -n "${commit}" | base64)
```
Comment (mbarison) [inline docs/topics/examples/simple-event/simple-event.sh]: you must add -w 0

File: docs/topics/gateways.md (hunk 1 above)
Comment (bacongobbler) [reply docs/topics/gateways.md]: Projects like [kashti](https://github.com/Azure/kashti) display the jobs in a UI. By using a ULID we can display them in a logical order (such as by job creation time).

File: docs/topics/gateways.md (hunk 2 above)
Comment (bacongobbler) [reply docs/topics/gateways.md]: We're using the term "build" to reference a "job", not necessarily to "build things".

File: docs/topics/examples/simple-event/simple-event.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  console.log("The system time is " + e.payload);
+});
+EOF
+)
+
+# Now we will generate a new event evrey 60 seconds.
+while : ; do
+  # We'll use a UUID instead of a ULID. But if you want a ULID generator, you
+  # can grab one here: https://github.com/technosophos/ulid
+  uuid="$(uuidgen | tr '[:upper:]' '[:lower:]')"
```
Comment (mbarison) [inline docs/topics/examples/simple-event/simple-event.sh]: if you use `uuidgen -t` then your UUID is time based, thus sortable.

File: docs/topics/examples/simple-event/simple-event.sh (hunk 3 above)
Comment (technosophos) [reply docs/topics/examples/simple-event/simple-event.sh]: Yeah, you are right. Alpine and Debian require -w 0 b/c their built-in wrap is 80.

File: docs/topics/examples/simple-event/simple-event.sh (hunk 4 above)
Comment (technosophos) [reply docs/topics/examples/simple-event/simple-event.sh]: My version of uuidgen does not have a `-t`. But you are right -- a UUIDv1 generator will make it at least sorta sortable.

File: docs/topics/examples/simple-event/simple-event.sh (hunk 3 above)
Comment (technosophos) [reply docs/topics/examples/simple-event/simple-event.sh]: Ugh... and macOS/BSD's base64 doesn't have a `-w`. It uses `-b` instead, and defaults to unwrapped. I'll do a little primitive OS checking to handle this.

**Comment by technosophos** [comment]
Okay, I think the new commit deals with all of the comments above. Notably, I do some OS detection now, and fixed some of the wording to be clearer about what certain things are doing.

**Pull request merged**